packages/web/src/services/audius-backend/WithdrawUSDC.ts

```diff
@@ -0,0 +1,224 @@
+      ASSOCIATED_TOKEN_PROGRAM_ID, // associatedProgramId
+      TOKEN_PROGRAM_ID, // programId
+      libs().solanaWeb3Manager!.mints.usdc, // mint
+      solanaUSDCATA, // associatedAccount
+      solanaRootAccount, // owner
+      feePayer // payer
+    )
+  return createSolanaUSDCAccountInstr
+}
+
+const closeUSDCATAInstruction = async (
+  solanaUSDCATA: PublicKey,
+  solanaRootAccount: PublicKey,
+  feePayer: PublicKey
+) => {
+  const closeSolanaUSDCAccountInstr = Token.createCloseAccountInstruction(
+    TOKEN_PROGRAM_ID, //    programId
+    solanaUSDCATA, //  account to close
+    feePayer, // fee destination
+    solanaRootAccount, //  owner
+    [] //  multiSigners
+  )
+  return closeSolanaUSDCAccountInstr
+}
+
+// Swaps USDC for SOL, taking USDC from the user's USDC user bank and
+// depositing SOL into the user's root solana account.
+const swapUSDCForSol = async (
+  amount: number,
+  solanaRootAccountPubkey: PublicKey,
+  usdcUserBank: PublicKey,
+  solanaUSDCATA: PublicKey,
+  feePayerPubkey: PublicKey
+) => {
+  const connection = await getSolanaConnection()
+
+  // TODO: double check slippage
+  const quoteRoute = await JupiterSingleton.getQuote({
+    inputTokenSymbol: 'USDC',
+    outputTokenSymbol: 'SOL',
+    inputAmount: amount,
+    slippage: USDC_SLIPPAGE,
+    swapMode: SwapMode.ExactOut,
+    onlyDirectRoutes: true
+  })
+  const usdcNeededAmount = quoteRoute.inputAmount
+
+  const swapRoute = await JupiterSingleton.getQuote({
+    inputTokenSymbol: 'USDC',
+    outputTokenSymbol: 'SOL',
+    inputAmount: usdcNeededAmount.uiAmount,
+    slippage: USDC_SLIPPAGE,
+    onlyDirectRoutes: true
+  })
+  const exchangeInfo = await JupiterSingleton.exchange({
+    routeInfo: swapRoute.route,
+    userPublicKey: solanaRootAccountPubkey,
+    feeAccount: feePayerPubkey
+  })
+  const swapInstructions =
+    exchangeInfo.transactions.swapTransaction.instructions
+
+  const transferInstructions =
+    await libs().solanaWeb3Manager!.createTransferInstructionsFromCurrentUser({
+      amount: usdcNeededAmount.uiAmount,
+      feePayerKey: feePayerPubkey,
+      senderSolanaAddress: usdcUserBank,
+      recipientSolanaAddress: solanaUSDCATA,
+      instructionIndex: 1,
+      claimableTokenPDA: libs().solanaWeb3Manager!.claimableTokenPDAs.usdc,
+      solanaTokenProgramKey: TOKEN_PROGRAM_ID,
+      claimableTokenProgramKey:
+        libs().solanaWeb3Manager!.claimableTokenProgramKey,
+      connection,
+      mint: 'usdc'
+    })
+
+  const createATAInstruction = await createUSDCATAInstruction(
+    solanaUSDCATA,
+    solanaRootAccountPubkey,
+    feePayerPubkey
+  )
+  const closeATAInstruction = await closeUSDCATAInstruction(
+    solanaUSDCATA,
+    solanaRootAccountPubkey,
+    feePayerPubkey
+  )
+
+  const instructions = [
+    createATAInstruction,
+    ...transferInstructions,
+    ...swapInstructions,
+    closeATAInstruction
+  ]
+  return await libs().solanaWeb3Manager!.transactionHandler.handleTransaction({
```

Comment:
super nit: maybe this fn can return the transaction, and the saga can be responsible for sending it (with or without libs/transaction handler?)